javalib/src/main/scala/java/util/concurrent/Semaphore.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.util.concurrent
+
+import java.util.{Collection, Collections}
+
+class Semaphore(private[this] var permits: Int, fairness: Boolean) extends Serializable {
+
+  def this(permits: Int) = this(permits, false)
+
+  // These methods can't be implemented because they block
+  // def acquire(): Unit = ???
+  // def acquire(permits: Int): Unit = ???
+  // def acquireUninterruptibly(): Unit = this.acquire()
+  // def acquireUninterruptibly(permits: Int): Unit = this.acquire(permits)
+  //def tryAcquire(permits: Int, timeout: Long, unit: TimeUnit): Boolean =
+  // def tryAcquire(timeout: Long, unit: TimeUnit): Boolean =
+
+  def availablePermits(): Int = permits
+
+  def drainPermits(): Int = {
+    val old = permits
+    permits = 0
+    old
+  }
+
+  protected def getQueuedThreads(): Collection[Thread] = Collections.emptySet()
+
+  def getQueueLength(): Int = 0
+
+  def hasQueuedThreads(): Boolean = false
+
+  def isFair(): Boolean = fairness
+
+  protected def reducePermits(reduction: Int): Unit = {
+    permits = permits - reduction
+  }
+
+  def release(): Unit = release(1)
+
+  def release(permits: Int): Unit = {
+    if (permits < 0 ) {
```

Comment:
remove the space before the brace (`if (permits < 0)`). (this is what is failing scalastyle).